File: cosmos/ProConnect/src/app/Class/profileconf.ts
import { IProfileconf } from "../interfaces/iprofileconf";

export class Profileconf implements IProfileconf{
  id: number = 0;
  working: boolean = false;
  homeoffice: boolean = false;
  salaryexpectation: number = 0;


  public getHomeoffice(): boolean {return this.homeoffice  }
  public setHomeoffice(homeoffice: boolean): void { this.homeoffice = homeoffice; }

  public getWorking(): boolean { return this.working}
  public setWorking(working: boolean): void { this.working = working; }

  public getSalaryexpectation(): number { return this.salaryexpectation  }
  public setSalaryexpectation(salaryexpectation: number): void { this.salaryexpectation = salaryexpectation; }

  public getId(): number { return this.id}
  public setId(id: number): void { this.id = id; }
}

File: cosmos/Cosmos-API/src/main/java/com/planbtech/cosmos/model/entites/ProfileConf.java
package com.planbtech.cosmos.model.entites;

import com.planbtech.cosmos.model.Interfaces.IProfileConf;
import jakarta.persistence.*;

@Entity(name = "profileconf")
public class ProfileConf implements IProfileConf {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "profileconf_id", nullable = false)
    private Long id;

    @Column(name = "working")
    private boolean working;

    @Column(name = "homeOffice")
    private boolean homeoffice;

    @Column(name = "salaryexpectation")
    private float salaryExpectation;

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public void setWorking(boolean working) {
        this.working = working;
    }

    @Override
    public boolean getWorking() {
        return this.working;
    }

    @Override
    public void setHomeoffice(boolean homeoffice) {
        this.homeoffice = homeoffice;
    }

    @Override
    public boolean getHomeoffice() {
        return this.homeoffice;
    }

    @Override
    public void setSalaryExpectation(float salaryExpectation) {
        this.salaryExpectation = salaryExpectation;
    }

    @Override
    public float getSalaryExpectation() {
        return this.salaryExpectation;
    }
}
